src/codegen/opt/build_cuda_on.cc

```diff
@@ -26,11 +28,50 @@ namespace codegen {
     }                                                                   \
   }
 
-std::string NVRTCCompile(const std::string& code) {
+std::string NVRTCCompile(const std::string& code, bool include_path = false) {
+  char *compileParams[2];
+  int numCompileOptions = 0;
   nvrtcProgram prog;
+  cudaDeviceProp deviceProp;
+  std::string cc = "30";
+  cudaError_t e = cudaGetDeviceProperties(&deviceProp, 0);
+
+  if (e == cudaSuccess) {
+    cc = std::to_string(deviceProp.major) + std::to_string(deviceProp.minor);
+  } else {
+    LOG(WARNING) << "cannot detect compute capability from your device, "
+                 << "fall back to compute_30.";
+  }
+
+  std::string archOption = "-arch=compute_" + cc;
+  compileParams[numCompileOptions] = reinterpret_cast<char *>(malloc(sizeof(char) *
+                                                                     (archOption.length() + 1)));
+  snprintf(compileParams[numCompileOptions], archOption.length() + 1, "%s", archOption.c_str());
+  numCompileOptions++;
+
+  if (include_path) {
+    std::string includeOption = "--include-path=";
+    const char* cudaHomePath = std::getenv("CUDA_HOME");
```

Comment:
How about defining CUDA_HOME as a preprocessor macro in the .cmake file?

Reply:
Does it mean that `cudaHomePath` defined as a preprocessor macro at the time of building tvm? If so, I'm worried that its path depends strongly on the environment at the time of building tvm. For example, whether it will be a problem when distributing a pre-compiled tvm in the future. Please let me know your opinion.

Reply:
Okay, never mind.

Then, I'd suggest using CUDA_PATH instead of CUDA_HOME for the environment variable, and defaulting to /usr/local/cuda.  I think it's better to be consistent with how python/tvm/contrib/nvcc.py finds the cuda path.

Reply:
@kazum Thank you for the suggestion. I'll address it.